source/util.cpp

```diff
@@ -99,20 +100,35 @@ string indent(string const &code, string const &indentation)
 	return r;
 }
 
+std::string join( std::vector<std::string>& splits, std::string const& sep ) {
+	if ( !splits.empty() ) {
+		std::ostringstream ss;
+		ss << splits[0];
+		for ( unsigned j=1; j<splits.size(); ++j ) {
+			ss << sep << splits[j];
+		}
+  	return ss.str();
+	}
+	return std::string();
```

Comment:
I think this could be simplified a bit, how about:
```C++
std::string join( std::vector<std::string> const &splits, std::string const& sep )
{
	std::string r;
	for(auto const & s : splits) { r += s; r+= sep; }
	if( r.size() ) r.resize(r.size() - sep.size());
	return r;
}
```

Reply:
In general, std::ostringstream is optimized for appends, which std::string in itself is not. What you are proposing is way more expensive and would have a big impact in large batches in general. In this case, not very much.
As far as simplifying, as I read it looks more complex but that's just me.

Reply:
> In general, std::ostringstream is optimized for appends

-- Thank you for info @HFTrader ! Interesting, i was not aware of this. Could you please point me where this is specified? Because i was under impression that `std::ostringstream` is just a wrapper on `std::string` which is used as a buffer. For instance, cppreference.com is explicitly stating it here: https://en.cppreference.com/w/cpp/io/basic_ostringstream . Thanks,